app/app.cpp

```diff
@@ -1,12 +1,49 @@
 #include "app.h"
 
-#include <format>
+namespace vsite::oop::p1 
+{
+	double f1(int a, int b)
+	{
+		return pow(a, b);
+	}
 
-namespace vsite::oop::p1 {
+	bool f2(bool a)
+	{
+		return !a;
+	}
 
-/*
-* function implementations
-*/
+	int f3(int a, int b, int c, int d)
+	{
+		return a + b + c + d;
+	}
 
+	std::string f4(int a)
+	{
+		return "int";
+	}
 
-} // namespace
+	std::string f4(bool a)
+	{
+		return "bool";
+	}
+
+	std::string f4(double a)
+	{
+		return "double";
+	}
+
+	bool f5(std::string& s, int a)
+	{
+		if (a > 0)
+		{
+			s += std::to_string(a % 10);
+			return true;
+		}
+
+		return false;
+	}
+
+	void f6(int a, int b, std::stringstream& ss) {
```

Comment:
Ne, ideja je da funkcija prima `std::ostream&` - bilo koji output stream (što onda može biti `std::cout`, ali i `std::stringstream` za unit testove).